source/common/buffer/buffer_impl.cc

```diff
@@ -183,5 +183,22 @@ OwnedImpl::OwnedImpl(const Instance& data) : OwnedImpl() { add(data); }
 
 OwnedImpl::OwnedImpl(const void* data, uint64_t size) : OwnedImpl() { add(data, size); }
 
+std::string OwnedImpl::toString() const {
```

Comment:
How is this different to `linearize()`? Do we need both variants?

Reply:
I looked at linearize and though they seem similar, I don't see how either can be efficiently implemented on the other.

Reply:
linearize mutates the buffer IIUC; I thought it made sense to have a simple observer that can returned the fully flattened string.

Moreover it exists in the codebase now, in a test helper, and I'm mostly just moving it. However per your other comment -- I don't really need this in the interface, as everywhere in tests and in admin.cc where it's needed, I have an OwnedImpl. Removed.

Reply:
I think mutating is a benefit here, it's kind of like defragging. At least naively, it seems you should be able to implement it about as efficiently as the `toString()` operation.